src/Testcontainers.LocalStack/LocalStackBuilder.cs

```diff
@@ -0,0 +1,102 @@
+namespace Testcontainers.LocalStack;
+
+/// <inheritdoc cref="ContainerBuilder{TBuilderEntity, TContainerEntity, TConfigurationEntity}" />
+[PublicAPI]
+public sealed class LocalStackBuilder : ContainerBuilder<LocalStackBuilder, LocalStackContainer, LocalStackConfiguration>
+{
+    public const ushort LocalStackPort = 4566;
+    public const string LocalStackImage = "localstack/localstack:1.3.1";
+    
+    /// <summary>
+    /// Initializes a new instance of the <see cref="LocalStackBuilder" /> class.
+    /// </summary>
+    public LocalStackBuilder()
+        : this(new LocalStackConfiguration())
+    {
+        DockerResourceConfiguration = Init().DockerResourceConfiguration;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="LocalStackBuilder" /> class.
+    /// </summary>
+    /// <param name="dockerResourceConfiguration">The Docker resource configuration.</param>
+    private LocalStackBuilder(LocalStackConfiguration dockerResourceConfiguration)
+        : base(dockerResourceConfiguration)
+    {
+        DockerResourceConfiguration = dockerResourceConfiguration;
+    }
+
+    /// <inheritdoc />
+    protected override LocalStackConfiguration DockerResourceConfiguration { get; }
+    
+    /// <summary>
+    /// Sets the LocalStack external service start port.
+    /// </summary>
+    /// <param name="port">The LocalStack external service start port.</param>
+    /// <returns>A configured instance of <see cref="LocalStackBuilder" />.</returns>
+    public LocalStackBuilder WithExternalServicePortStart(string port)
+    {
+        return Merge(DockerResourceConfiguration, new LocalStackConfiguration(externalServicePortStart: port))
+            .WithEnvironment("EXTERNAL_SERVICE_PORTS_START", port);
+    }
+    
+    /// <summary>
+    /// Sets the LocalStack external service end port.
+    /// </summary>
+    /// <param name="port">The LocalStack external service end port.</param>
+    /// <returns>A configured instance of <see cref="LocalStackBuilder" />.</returns>
+    public LocalStackBuilder WithExternalServicePortEnd(string port)
+    {
+        return Merge(DockerResourceConfiguration, new LocalStackConfiguration(externalServicePortEnd: port))
+            .WithEnvironment("EXTERNAL_SERVICE_PORTS_END", port);
+    }
+
+    /// <inheritdoc />
+    public override LocalStackContainer Build()
+    {
+        Validate();
+        return new LocalStackContainer(DockerResourceConfiguration, TestcontainersSettings.Logger);
+    }
+
+    protected override LocalStackBuilder Init()
+    {
+        return base.Init()
+            .WithImage(LocalStackImage)
+            .WithPortBinding(LocalStackPort, true)
+            .WithExternalServicePortStart("4510")
+            .WithExternalServicePortEnd("4559")
+            .WithWaitStrategy(Wait.ForUnixContainer().UntilHttpRequestIsSucceeded(request => request.ForPath("/_localstack/health").ForPort(LocalStackPort)));
```

Comment:
I am not very familiar with LocalStack, how likely developers will use an older version (overrides the image) that does not use the health check endpoint? Does it even make sense to use and older version?

Reply:
In Java, the wait strategy waits for a log entry. See https://github.com/testcontainers/testcontainers-java/blob/343c5a533f4c6148d6e6eea3a3374116365bc91d/modules/localstack/src/main/java/org/testcontainers/containers/localstack/LocalStackContainer.java#L98, we adopted it for the Go implementation (https://github.com/testcontainers/testcontainers-go/pull/800/files#diff-bd71ec18d411b5b0b51723761d6d0a95e60ad7d5578828df544e14d6518a222dR53) and suggested it for Rust https://github.com/testcontainers/testcontainers-rs/pull/462#discussion_r1102602196

Reply:
To be fair, I do prefer having the wait for an HTTP endpoint, as it defines a more strict contract than waiting for a log entry. Probably something to discuss with the other languages cc/ @eddumelendez @kiview

Reply:
> I am not very familiar with LocalStack, how likely developers will use an older version (overrides the image) that does not use the health check endpoint? Does it even make sense to use and older version?

It may make sense to use an older version if some feature gets removed or when trying to find a bug.

Wait for a log entry may be an option, but I would benchmark both the health check and log entry strategies and go with the fastest. I find the speed more important than the specific scenario of someone overriding to an older version.

Other option would be to create both strategies and a specific method to override the default wait strategy. SOmething like this:

```
.WithWaitStrategy(LocalStackWaitStrategy.WaitForHealthCheck)
```

Reply:
I think we favor the health check 😬. We are just not sure if it makes sense to use an older version of LocalStack. I would imagine not, who wants to test against an older version of a cloud product?

Reply:
ok no problem i just revert my previous changes :)